src/test/java/com/google/devtools/build/lib/remote/RemoteSpawnCacheTest.java

```diff
@@ -369,6 +405,26 @@ public void noCacheSpawns() throws Exception {
     }
   }
 
+  @Test
+  public void noRemoteCacheStillUsesLocalCache() throws Exception {
```

Comment:
I just though of another test case: what if we have combined cache - remote (http for now) and local (disk) - AND `no-remote-cache` tag. For now, bazel would still be using both caches, but should use **only** disk cache in such a case.
I think we missed this use case. Could you please add test cases and maybe code changes for that?

Reply:
Ah good catch. Yeah I'll add that

Reply:
I'm not sure what the best way to accomplish this would be. We don't have a way of disabling the remote part of the cache without disabling the whole thing from here, and the combined disk cache has no way of knowing the execution requirements (unless we pipe that information through the `AbstractRemoteActionCache` and `SimpleBlobStore`--but this feels very messy and it would break some of the abstraction). 

Would a reasonable approach be to treat a Combined cache as if it were a remote cache and disable it entirely if the `no-remote-cache` tag is present? This feels a bit unintuitive, but it'd be better than the current state (using both caches despite `no-remote-cache` being specified.

Reply:
hm, true. Okay, let's make like you've suggested - treat Combined Cache as a remote cache and disable it entirely if `no-remote-cache` tag is set.
I'd say, we need to also update docs about this case - to make it clear, that `no-remote-cache` tag  disables the Combined Cache as well.

Reply:
Done :)